portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -25,17 +34,59 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        
+        // Create Comment from request.
+        Comment c = Comment.makeComment(request);
+
+        // Set properties in Datastore entity.
+        Entity commentEntity = new Entity("comment");
+        commentEntity.setProperty("timestamp", c.getTimestamp());
+        commentEntity.setProperty("author", c.getAuthor());
+        commentEntity.setProperty("text", c.getText());
+        datastore.put(commentEntity);
+
+        // Return the comment ID.
+        Gson gson = new Gson();
+        response.setContentType("text/plain");
+        response.getWriter().println(commentEntity.getKey().getId());
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        
+        // Get comment limit parameter.
+        int max = Integer.parseInt(request.getParameter("max"));
```

Comment:
To consider: what happens if the user input passes in a non-number that gets plumbed to the backend?

Reply:
I am using a select menu for this as shown below:
![image](https://user-images.githubusercontent.com/14950581/84825200-f023e300-afd5-11ea-8670-d133f79a581c.png)
Because of this, the user shouldn't be able to pass in a nonnumber. Would it be better to check anyway?

Reply:
Added input validation anyway.